prod_chemical_release/scripts/combine_stewi_files.R

```diff
@@ -1,89 +1,216 @@
+    }
+  },
+  error=function(cond){ message("Error message: ", cond); return(NULL) },
+  finally={ dbDisconnect(con) })
+  return(query_result)
+}
 
-#'@description A function to create a connection to the prod_factotum database using .Renviron parameters
+#'@description A helper function to send a statement to the database to update table entries. 
+#'Handles errors/warnings with tryCatch.
+#'@param statement A SQL query string to send to the database
+#'@param con_type Whether to connect to postgres, mysql, or sqlite version
+#'@param schema The schema name to use if using a postgresql connection
+#'@import DBI dplyr magrittr
+#'@return None. A SQL statement is passed to the database to make changes to the database.
+send_statement_db <- function(statement=NULL, con_type=NULL, schema){
+  if(is.null(statement)) return(message("Must provide a statement to send"))
+  con = connect_to_db(con_type=con_type)
+  tryCatch({
+    if(con_type == "postgres"){#Add schema tag
+      return(dbSendStatement(con, statement %>% 
+                               gsub("FROM ", "FROM ",schema,".", .) %>%
+                               gsub("INTO ", "INTO ",schema,".", .) %>%
+                               gsub("UPDATE ", "UPDATE ",schema,".", .) %>%
+                               gsub("EXISTS ", "EXISTS ",schema,".", .)
+      ) %>% dbClearResult())
+    } else {
+      return(dbSendStatement(con, statement) %>% dbClearResult())
+    }
+  },
+  error=function(cond){ message("Error message: ", cond); return(NA) },
+  warning=function(cond){ message("Warning message: ", cond); return(NULL) },
+  finally={ dbDisconnect(con) })
+}
+
+#'@description A helper function to write a table to the datbase. Note, if it already exists, 
+#'the table will be overwritten.
+#'@param name The name of the table to write to the database
+#'@param data The data to push into the new database table
+#'@param con_type Whether to connect to postgres, mysql, or sqlite version
+#'@param schema The schema name to use if using a postgresql connection
 #'@import DBI
-#'@return A mysql connection object
-connect_to_db <- function(){
-  return(dbConnect(RMySQL::MySQL(), 
-                   username = Sys.getenv("user"), password = Sys.getenv("pass"), 
-                   host = "mysql-ip-m.epa.gov", port = 3306, dbname = "prod_chemical_release"))
+#'@return None. A table is written (or overwritten) on the database with a given 'name' and 'data'.
+write_table_db <- function(name=NULL, data=NULL, con_type=NULL){
+  if(is.null(name)) return(message("Must provide a name for the database table"))
+  if(is.null(data)) return(message("Must provide data to push to the database table"))
+  con = connect_to_db(con_type)
+  tryCatch({
+    if(con_type == "postgres"){
+      dbWriteTable(con, name =c(schema, name), value = data, row.names = FALSE, overwrite = TRUE)
+    } else {
+      #Make sure use of local files to write data is allowed
+      dbSendQuery(con, "SET GLOBAL local_infile = true;")

```

Comment:
Need to check if this will actually work. I know certain "SET" statements only work within the same connections session and become reset across connections. 

If so, we'd just need to write a tryCatch that makes a connection, then run the sendQuery and writeTable functions without the helper functions, with a finally to disconnect.